addons/account/models/account_move.py

```diff
@@ -4090,6 +4090,13 @@ def action_transfer_accounts_wizard(self):
             'context': self.env.context,
         }
 
+    def _get_attachment_search_parameters(self):
+        search_parameters = [{'model': 'account.move', 'id': self.move_id.id}]
+        if self.statement_id:
+            search_parameters.append({'model': 'account.bank.statement', 'id': self.statement_id.id})
+        if self.payment_id:
+            search_parameters.append({'model': 'account.payment', 'id': self.payment_id.id})
+        return search_parameters
```

Comment:
why not:

```
def _get_attachment_domains(self):
  domains = [('res_model', '=', 'account.move'), ('res_id', '=', self.move_id.id)]
  if self.statement_id:
    domains = [('res_model', '=', 'account.bank.statement'), ('res_id', '=', self.statement_id.id)]
  if self.payment_id:
    domains = [('res_model', '=', 'account.payment'), ('res_id', '=', self.payment_id.id)]
  return domains
```

Later, you will make something like:
`attachments = self.env['ir.attachment'].search(expression.OR(self._get_attachment_domains()))`

Reply:
TIL expression.OR